room/room-compiler/src/main/kotlin/androidx/room/processor/MethodProcessorDelegate.kt

```diff
@@ -71,6 +76,31 @@ abstract class MethodProcessorDelegate(
             ).process()
         }
     }
+    fun isUsingMultipleConcurrencyPatterns(): Boolean {
```

Comment:
I would move this to a top-level extension function on `MethodProcessorDelegate`, the first check should be for suspend function and return early and then go through the list doing `isAssignableFrom`. If we can return early we should because finding types and checking for assignability is expensive.

Reply:
Done